### DISCUSSION TITLE
Cookie not working for me using Annotated type.

### DISCUSSION THREAD

**Discussion by milsman2** [fastapi/fastapi#9325] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import APIRouter, Cookie, Response


app = FastAPI()


@app.get(
    "/read_cookie/",
    status_code=200,
)
async def read_cookie(fakesession: Annotated[str | None, Cookie()] = None):
    return {"fakesession": fakesession}


@app.post("/cookie-and-object/")
def create_cookie(response: Response):
    response.set_cookie(key="fakesession", value="fake-cookie-session-value")
    return {"message": "Come to the dark side, we have cookies"}
```

### Description

* Called endpoint from Postman to eliminate possible cause from SwaggerUI.
* async def read_cookie(fakesession: str | None = Cookie(None)): as the function seems to work fine.
* Using example code submitted retuns {"fakesession": null}

### Operating System

Linux

### FastAPI Version

0.92.0

### Python Version

Python 3.10.6

**Comment by n8sty** [comment, marked as answer]
Same issue as discussed/reported here: https://github.com/tiangolo/fastapi/discussions/9279
